### ISSUE TITLE
Key priority setting doesn't compare individually mapped keys to key combinations

### ISSUE THREAD

**Issue by TakuikaNinja** [TASEmulators/BizHawk#3842]
[//]: # "This description supports Markdown syntax. There's a cheatsheet here: https://guides.github.com/features/mastering-markdown/"
[//]: # "These lines are comments, for letting you know what you should be writing. You can delete them or leave them in."
[//]: # "Also, please don't waste your time writing until you've checked for similar Issues. Remember to check closed Issues too!"
[//]: # "One more thing: if you're on Linux, please comment on #1430 instead of opening an issue so we don't annoy the other devs."

### Summary
[//]: # "Briefly describe what's broken. Include relevant details: loaded core, loaded rom's hash, open tools, running scripts... You can embed a screenshot if it's easier to show the bug, but if you need more than one please put them at the end."
If keys mapped to separate inputs conflict with a hotkey combination, the hotkey is always prioritised regardless of the key priority setting. This is very counter-intuitive, especially on cores for computers where one may attempt to recreate the keyboard mapping from another emulator.

### Repro
[//]: # "If you can't figure out the list of steps, delete this section and put 'heisenbug' in the summary somewhere. If a Lua script can cause the bug, you can embed that instead (as simple as possible please)."
The steps here use ZXHawk as the issue is most apparent here. It should occur in other cores, too.
1. Load ZXHawk and a game in 48K mode, you should be put into a BASIC prompt
2. Set the key priority to "input overrides hotkeys"
3. In the controller settings, assign `Ctrl` and `O` to `Key Symbol Shift` and `Key O` respectively
4. Press the keys together - the open file hotkey (`Ctrl+O`) is recognised instead of the inputs (correct input should put `;` in the BASIC prompt as it's `Symbol Shift+O`)

### Host env.
[//]: # "List the computers you've found the bug with. If there's a version that doesn't have the bug, please put that in too. Here are some examples:"
The environment shouldn't matter as I've been told the input handling code is OS-agnostic. 
- BizHawk 2.9.1; EndeavourOS; AMD (Me)
- BizHawk 2.9.1; Windows (CasualPokePlayer)

[//]: # "(screenshots, if applicable)"

[//]: # "That's it! If you'd like to help more, you could try a dev build (see Testing in the readme) or an older release. Click submit now and you can edit it later."

**Comment by Morilli** [comment]
What's the exact intention here?
The given case seems simple enough, you're pressing 2 keys, both of which are configured as input, but there's also a hotkey that is the combination of the two, so the hotkey gets activated instead, as there is no input that exactly matches the combination.

We could probably check if all of the individually pressed keys are inputs and suppress the hotkey in this case. But what if only part of them are input? Let's say I have a controller button mapped to `A`, and a hotkey mapped to `Ctrl+A`. When I press `Ctrl+A`, should the `A` input still block the hotkey or should the hotkey be prioritised?

**Comment by TakuikaNinja** [comment]
Most emulators with keyboard inputs avoid conflicting inputs in the first place. Fuse (the ZX Spectrum emulator I was basing the input mapping on) has its "open file" hotkey set to a key that would never be triggered using keystrokes meant for the emulated machine (F3 in this case). Another example is Mesen, which disables shortcut keys entirely while a keyboard peripheral is being emulated. Both of these emulators hardcode the keyboard mapping (likely with some layout detection), as with many other emulators for keyboard-based systems. 

The default input mapping for computer cores in Bizhawk contain nothing for the keyboard - only the joystick is mapped to something sensible. I presume this is because setting a default keyboard mapping would conflict with the default hotkeys. This results in users like myself having to:
- manually map all of the keys
- then change the hotkeys to prevent conflicts

I didn't like doing that for each and every computer core I want to use, so this is how I ended up here. Solving the underlying conflict would allow for default keyboard mappings to be created in the future, which would be more user-friendly in my opinion. 

> But what if only part of them are input? Let's say I have a controller button mapped to `A`, and a hotkey mapped to `Ctrl+A`. When I press `Ctrl+A`, should the `A` input still block the hotkey or should the hotkey be prioritised?

I believe the key priority setting should determine the outcome. If inputs override hotkeys, then the hotkey should be suppressed. If hotkeys override inputs, then the hotkey should be prioritised instead. As for "both inputs and hotkeys", having both occur seems sensible but may result in confusing behaviour. The exact implementation can be discussed and altered if needed. 

The bottom line is that the current behaviour of hotkeys overriding partial inputs breaks the expectations set by the key priority setting.

**Comment by Morilli** [comment]
The key priority part should be fixed. As for your other suggestions, like default keyboard mappings, that might work as an own issue.
The default controller mappings in general aren't updated often, so that might be doable here.
